crates/config/src/lib.rs

```diff
@@ -4975,4 +5001,44 @@ mod tests {
             Ok(())
         });
     }
+
+    #[traced_test]
+    #[test]
+    fn warn_unknown_keys_logs_warning_for_unknown_keys() {
```

Comment:
these are good but would be great to have an e2e test too where we run forge for a project with unknown config keys defined in foundry.toml and inline and we assert proper warnings displayed